content/6_Advanced/Extend_Euclid.mdx

```diff
@@ -34,6 +34,36 @@ ll euclid(ll a, ll b) {
 }
 ```
 
+</CPPSection>
+<PySection>
+
+```py
+def euclid(a: int, b: int) -> int:
+    """Returns integer representing GCD of two positive, non-zero numeric values.
+    """
+
+    assert int(a) > 0 and int(b) > 0, \
+        "Arguments must be positive, non-zero numeric values."
+	
+    def swap(x, y):
+        tmp = x
+        x = y
+        y = tmp
+        return x, y
+
+    while b > 0: # run Euclidean algo
+        k = a // b
+	# subtract multiples of one equation from the other.
+        a -= b * k
+        a, b = swap(a, b)
```

Comment:
```suggestion
        a, b = b, a
```

Reply:
Good suggestion! I understand how to swap without a function, but I was trying to stay close to the C++ implementation.

How many approvers are on this pull request?

Reply:
Currently 1